react/components/SearchBar.js

```diff
@@ -7,42 +7,43 @@ import { Button } from 'vtex.styleguide'
 
 import header from '../store-header.css'
 
-const SearchBar = ({ compactMode, autoFocus, onCancel }) => {
-  return (
-    <React.Fragment>
-      <div className={`${header.topMenuSearchBar} flex pa2-m flex-grow-1 justify-center`}>
-        <div className="w-100 mw7">
-          <Adopt
-            mapper={{
-              placeholder: <FormattedMessage id='header.search-placeholder' />,
-              emptyPlaceholder: <FormattedMessage id='header.search-emptyPlaceholder' />,
-            }}
-          >
-            {({ placeholder, emptyPlaceholder }) => (
-              <ExtensionPoint
-                id="search-bar"
-                placeholder={placeholder}
-                emptyPlaceholder={emptyPlaceholder}
-                compactMode={compactMode}
-                autoFocus={autoFocus}
-              />
-            )}
-          </Adopt>
-        </div>
-      </div>
-      {compactMode && (
+const SearchBar = ({ isMobile, autoFocus, onCancel, height, iconClasses }) => (
+  <div className={`${header.topMenuSearchBar} flex pa2-m flex-grow-1 justify-between`} style={{height: height}}>
+    <div className="w-75">
+      <Adopt
+        mapper={{
+          placeholder: <FormattedMessage id='header.search-placeholder' />,
+          emptyPlaceholder: <FormattedMessage id='header.search-emptyPlaceholder' />,
+        }}
+      >
+        {({ placeholder, emptyPlaceholder }) => (
+          <ExtensionPoint
+            id="search-bar"
+            placeholder={placeholder}
+            emptyPlaceholder={emptyPlaceholder}
+            autoFocus={autoFocus}
+            hasIconLeft={isMobile}
+            iconClasses={iconClasses}
+          />
+        )}
+      </Adopt>
+    </div>
+    {isMobile && (
+      <div className="w-25 pa2-m pt2-s">
         <Button size="small" variation="tertiary" onClick={onCancel}>
-          <FormattedMessage id="header.search-cancel" />
+          <span className='near-black ttl'><FormattedMessage id="header.search-cancel" /></span>
         </Button>
-      )}
-    </React.Fragment>
-  )
-}
+      </div>
+    )}
+  </div>
+)
 
 SearchBar.propTypes = {
   onCancel: PropTypes.func,
-  compactMode: PropTypes.bool,
+  isMobile: PropTypes.bool,
```

Comment:
I think that injecting the `runtime.hints.mobile` in whichever component that really uses it is better than prop drilling, but it is up to you.